src/main/clojure/clara/rules/compiler.clj

```diff
@@ -235,15 +235,21 @@
   "Compiles a sequence of constraints into a structure that can be evaluated.
 
    Callers may also pass a collection of equality-only-variables, which instructs
-   this function to only do an equality check on them rather than create a unification binding."
+   this function to only do an equality check on them rather than create a unification binding.
+
+   The returned expression evaluates to a two item vector, [bindings error], which contains
+   the evaluated bindings in the first argument and a map with information on any exception
+   that might have been thrown by evaluating a constraint as the second vector entry."
```

Comment:
A variation of Option 1 would be to keep the outer, node-level try/catch in ClojureScript, and the fine-grained, per-expression try/catch in Clojure. This way ClojureScript users could at least see the entire expression that threw an exception (although not the specific constraint.)

I'd be in favor of that variation, or Will's Option 1 as he described it as well. I think it's okay that we offer slightly different error reporting on these platforms based on their capabilities, especially since the overall flow is the same.